aten/src/ATen/native/native_functions.yaml

```diff
@@ -7698,10 +7704,21 @@
 
 - func: diag.out(Tensor self, int diagonal=0, *, Tensor(a!) out) -> Tensor(a!)
   dispatch:
-    CPU, CUDA: diag_out
+    CPU: diag_cpu_out
+    CUDA: diag_cuda_out
+    MPS: diag_mps_out
 
 - func: diag(Tensor self, int diagonal=0) -> Tensor
```

Comment:
Is there a comment explaining the reasoning for the change to `diag`?

It looks like before, `diag` was a `CompositeImplicitAutograd` op that called `diag_embed`, and now we have a brand new backward op for it.

Reply:
This is a mistake. The diag improvements were added to master indepently and never added to the branch.
I'll make sure not to revert https://github.com/pytorch/pytorch/pull/87180 by mistake :p